pkg/api/norman/store/cluster/cluster_store.go

```diff
@@ -287,6 +288,14 @@ func loadDataFromTemplate(clusterTemplateRevision *v3.ClusterTemplateRevision, c
 
 	defaultedAnswers := make(map[string]string)
 
+	// The key in the map is used to preserve the order of registries
+	registryMap := make(map[int]map[string]interface{})
+	existingRegistries := convert.ToMapSlice(convert.ToMapInterface(existingCluster[managementv3.ClusterSpecFieldRancherKubernetesEngineConfig])[managementv3.RancherKubernetesEngineConfigFieldPrivateRegistries])
+	if existingRegistries != nil {
```

Comment:
Don't need this check. If `existingRegistries` is `nil`, then the `for` loop will be skipped (as if `existingRegistries` was empty).

Reply:
fixed.